server/internal/usecase/interactor/item.go

```diff
@@ -1179,3 +1187,207 @@ func (i Item) getReferencedItems(ctx context.Context, fields []*item.Field) ([]i
+		if err != nil {
+			return nil, err
+		}
+
+		schemaPackage, err := i.buildSchemaPackage(ctx, model)
+		if err != nil {
+			return nil, err
+		}
+
+		// fromPagination
+		paginationOffset := fromPagination(page, perPage)
+
+		items, _, err := i.repos.Item.FindBySchema(ctx, schemaPackage.Schema().ID(), nil, nil, paginationOffset)
+		if err != nil {
+			return nil, err
+		}
+
+		pr, pw := io.Pipe()
+		err = csvFromItems(pw, items, schemaPackage.Schema())
+		if err != nil {
+			return nil, err
+		}
+
+		return pr, nil
+	})
+}
+
+// ItemsWithProjectAsGeoJSON converts items content to Geo JSON given by project ID or project Alias and model ID or model Key
+func (i Item) ItemsWithProjectAsGeoJSON(ctx context.Context, projectIDorAlias project.IDOrAlias, modelIDOrKey model.IDOrKey, page *int, perPage *int, operator *usecase.Operator) (*integrationapi.FeatureCollection, error) {
+	if operator.AcOperator.User == nil && operator.Integration == nil {
+		return nil, interfaces.ErrInvalidOperator
+	}
+
+	return Run1(ctx, operator, i.repos, Usecase().Transaction(), func(ctx context.Context) (*integrationapi.FeatureCollection, error) {
+		project, err := i.repos.Project.FindByIDOrAlias(ctx, projectIDorAlias)
+		if err != nil {
+			return nil, err
+		}
+
+		model, err := i.repos.Model.FindByIDOrKey(ctx, project.ID(), modelIDOrKey)
+		if err != nil {
+			return nil, err
+		}
+
+		schemaPackage, err := i.buildSchemaPackage(ctx, model)
+		if err != nil {
+			return nil, err
+		}
+
+		// fromPagination
+		paginationOffset := fromPagination(page, perPage)
+
+		items, _, err := i.repos.Item.FindBySchema(ctx, schemaPackage.Schema().ID(), nil, nil, paginationOffset)
+		if err != nil {
+			return nil, err
+		}
+
+		featureCollections, err := featureCollectionFromItems(items, schemaPackage.Schema())
+		if err != nil {
+			return nil, err
+		}
+
+		return featureCollections, nil
+	})
+}
+
+func (i Item) buildSchemaPackage(ctx context.Context, model *model.Model) (*schema.Package, error) {
+	schemaIDs := id.SchemaIDList{model.Schema()}
+	if model.Metadata() != nil {
+		schemaIDs = append(schemaIDs, *model.Metadata())
+	}
+	schemaList, err := i.repos.Schema.FindByIDs(ctx, schemaIDs)
+	if err != nil {
+		return nil, err
+	}
+	s := schemaList.Schema(lo.ToPtr(model.Schema()))
+	if s == nil {
+		return nil, rerror.ErrNotFound
+	}
+
+	groups, err := i.repos.Group.FindByIDs(ctx, s.Groups())
+	if err != nil {
+		return nil, err
+	}
+
+	groupSchemaList, err := i.repos.Schema.FindByIDs(ctx, groups.SchemaIDs().Add(s.ReferencedSchemas()...))
+	if err != nil {
+		return nil, err
+	}
+	groupSchemaMap := lo.SliceToMap(groups, func(g *group.Group) (id.GroupID, *schema.Schema) {
+		return g.ID(), schemaList.Schema(lo.ToPtr(g.Schema()))
+	})
+	referencedSchemaMap := lo.Map(s.ReferencedSchemas(), func(s schema.ID, _ int) *schema.Schema {
+		return groupSchemaList.Schema(&s)
+	})
+
+	return schema.NewPackage(s, schemaList.Schema(model.Metadata()), groupSchemaMap, referencedSchemaMap), nil
+}
```

Comment:
```suggestion
```
get the schema package from uc.Schema.GetByModel in the adapter and pass it to the UC.
it's not good to duplicate the implementation, we can refactor it to a better way later.